app/models/supplier_location.rb

```diff
@@ -11,7 +11,7 @@ class SupplierLocation < ApplicationRecord
   scope :effective_to, ->(date) { where(effective_to: nil).or(where('effective_to >= ?', date)) }
   scope :effective_on, ->(date) { effective_from(date).effective_to(date) }
 
-  def self.link_locations(effective_from: nil, effective_to: nil, supplier:, locations:)
+  def self.link_locations(supplier:, locations:, effective_from: nil, effective_to: nil)
```

Comment:
Hm, not alphabetical then :D

Reply:
oh, ignore my last comment... I don't understand this rule?

Reply:
ah - they must be alphabetical, but mandatory params must come before optional params